routers/api/v1/repo/issue.go

```diff
@@ -176,7 +175,7 @@ func SearchIssues(ctx *context.APIContext) {
 
 	// Only fetch the issues if we either don't have a keyword or the search returned issues
 	// This would otherwise return all issues if no issues were found by the search.
-	if len(keyword) == 0 || len(issueIDs) > 0 || len(labelIDs) > 0 {
+	if len(keyword) == 0 || len(issueIDs) > 0 || len(includedLabelNames) > 0 {
```

Comment:
This maybe a bug but not a refactor?

Reply:
didn't dig into it yet - posible

Reply:
look's like in an edge case the label filter is not applied, but I could not hit it, so It's a code improvement to prevent becoming a bug  in the future